### PR TITLE
Update Develocity Gradle Plugin to 3.18.1

### PR DESCRIPTION
**Pull request by welandaz**
This PR updates Develocity Gradle Plugin to the latest `3.18.1` version

### CODE REVIEW

**Comment by welandaz** [comment]
@bot-gradle test this

**Review by ljacomet — CHANGES_REQUESTED** [review]
❌ `release` is not an active branch for the moment.
This should be merged to `master` instead.

File: build-logic-commons/build-platform/build.gradle.kts [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies {
     constraints {
         api("org.gradle.guides:gradle-guides-plugin:0.23")
         api("org.apache.ant:ant:1.10.14") // Bump the version brought in transitively by gradle-guides-plugin
-        api("com.gradle:develocity-gradle-plugin:3.17.6") // Run `build-logic-settings/update-develocity-plugin-version.sh <new-version>` to update
+        api("com.gradle:develocity-gradle-plugin:3.18.1") // Run `build-logic-settings/update-develocity-plugin-version.sh <new-version>` to update
```
Comment (wolfs) [inline build-logic-commons/build-platform/build.gradle.kts:28]: 🤔 The 8.10.1 patch release just went out, I think we shouldn't change the bundled version for 8.10 anymore. Let's do the PR on `master`.

File: build-logic-commons/gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -14,7 +14,7 @@ java {
 }
 
 dependencies {
-    compileOnly("com.gradle:develocity-gradle-plugin:3.17.6")
+    compileOnly("com.gradle:develocity-gradle-plugin:3.18.1")
```
Comment (wolfs) [inline build-logic-commons/gradle-plugin/build.gradle.kts:17]: ❌ This version should be removed, as on `master`.

**Review by wolfs — CHANGES_REQUESTED** [review]

**Comment by welandaz** [comment]
> ❌ `release` is not an active branch for the moment. This should be merged to `master` instead.
> 
> Even if we planned a 8.10.2, I am not sure we would take this change in it.

You mean release will be recreated from `master`? (e.g. there is going to be no merge to `master` if this PR ends in `release`)

**Comment by welandaz** [comment]
Closed in favor of https://github.com/gradle/gradle/pull/30491 as discussed in comments

**Pull request closed without merging**